Client/game_sa/CPoolsSA.cpp

```diff
@@ -1393,14 +1393,17 @@ int CPointerNodeSingleLinkPoolSA::GetNumberOfUsedSpaces()
     return iOut;
 }
 
-unsigned int CPoolsSA::AddTextureDictonarySlot()
+unsigned int CPoolsSA::AddTextureDictonarySlot(std::string strTxdName)
 {
     CTextureDictonarySAInterface* pTxd = (*m_ppTxdPoolInterface)->Allocate();
     if (!pTxd)
         return -1;
 
+    typedef uint(__cdecl * Function_TxdGatHashFromName)(char* strName);
+    Function_TxdGatHashFromName getHash = (Function_TxdGatHashFromName)(0x53CF30);
+
     pTxd->usUsagesCount = 0;
-    pTxd->hash = 0;
+    pTxd->hash = getHash(&strTxdName.at(0));
```

Comment:
I think this should be `.c_str()` instead?

Reply:
This is only draft and some API still in progress.